### ISSUE TITLE
AI doesn't switch on disadvantageous matchups and doesn't check for bad moves?

### ISSUE THREAD

**Issue by Syreldar** [DizzyEggg/pokeemerald#290]
I'm a Combusken vs a Lileep
Lileep knows ancient power, stealth rock, recover and giga drain
he starts off with stealth rock, which is good, and then spams ancient power
after 5 uses, ancient power finishes pp
at this point it should switch lileep for another pokemon which has advantage over combusken (it should do it after stealth rock instead of spamming ancient power tbh but yeah)
but instead keeps spamming recover even though lileep is at full hp, or uses giga drain
basically the trainer has no good moves left, yet doesn't switch for a better pokemon

**Comment by segabl** [comment]
from what I've seen the AI only switches when the score for all moves is below some value. The score for ancient power was probably still high, even though it was out of PP or the scores for all moves simply weren't low enough. I think this is a base game problem, the "switch if moves are bad" function hardly changed from vanilla iirc.

**Comment by ExoryxHana** [comment]
Score should be 0 (or whatever the null value for the score is) if the move is out of pp, that's just common sense.

**Comment by segabl** [comment]
The AI doesn't care about your mon having a super effective move. It only knows your moves once you used them and even then it only switches if its own moves are bad enough. Not saying you're wrong but that's just how the vanilla AI works.

**Comment by ExoryxHana** [comment]
I know, but it doesn't matter since the purpose of the bev2 is also to improve that aspect, that's why i'm reporting this.

**Comment by DizzyEggg** [comment]
Moved to RHH
